nowcasting_dataset/data_sources/pv/pv_data_source.py

```diff
@@ -98,11 +119,34 @@ def _load_metadata(self):
 
     def _load_pv_power(self):
 
-        logger.debug(f"Loading PV Power data from {self.filename}")
+        logger.debug(f"Loading PV Power data from {self.filenames}")
 
-        pv_power = load_solar_pv_data(
-            self.filename, start_dt=self.start_datetime, end_dt=self.end_datetime
-        )
+        # collect all metadata together
+        pv_power_all = []
+        for i, filename in enumerate(self.filenames):
+
+            # get pv power data
+            pv_power = load_solar_pv_data(
+                filename, start_dt=self.start_datetime, end_dt=self.end_datetime
+            )
+
+            # add i to columns, to make sure the columns are unique
+            new_columns = pv_power.columns.astype(str)
+            new_columns = new_columns + f"_{i}"
+            pv_power.columns = new_columns
+
+            pv_power_all.append(pv_power)
+
+        # join together pv power. Index is time, columns are pv ids. Make sure the ids are unique
+        pv_power = pv_power_all.pop(0)
```

Comment:
Sorry if I've misunderstood :slightly_smiling_face: but could lines 141 to 149 be replaced with something like:

```python
pv_power = pd.concat(pv_power_all, axis="columns")
assert not pv_power.columns.duplicated().any()
```

Reply:
yes, i think that is a better way, let me try it